File: problems/1476-count-negative-numbers-in-a-sorted-matrix/solution.java
class Solution {
    public int countNegatives(int[][] grid) {
        int counter = 0;
        for (int row = 0; row < grid.length; row++) {
            for (int col = grid[row].length - 1; col >= 0; col--) {
                if (grid[row][col] < 0){
                    counter ++;
                }
                else {
                    break;
                }
            }
        }
        return counter;
    }
}

File: problems/1476-count-negative-numbers-in-a-sorted-matrix/solution.go
func countNegatives(grid [][]int) int {
    row, col := 0, len(grid[0]) - 1
    res := 0

    for row < len(grid) && col >= 0 {
        if grid[row][col] >= 0 {
            row++
        } else {
            res += len(grid) - row
            col--
        }
    }
    
    return res
}